### DISCUSSION TITLE
perception of pydantic vs self-perception of pydantic

### DISCUSSION THREAD

**Discussion by srkunze** [fastapi/fastapi#8843] [Questions]
I am fairly new to both libraries, so please bear with me.

https://pydantic-docs.helpmanual.io/usage/models/ vs https://fastapi.tiangolo.com/python-types/#pydantic-models

One says:

> Pydantic is a Python library to perform data validation.

the other one says:

> pydantic is primarily a parsing library, not a validation library. Validation is a means to an end: building a model which conforms to the types and constraints provided.

As a newcomer, I would like to say: I am somewhat confused here. Could the documentation be amended to reflect these choices?

@samuelcolvin

**Comment by Hultner** [comment]
@srkunze While I agree that clarity in documentation is important, I think this issue actually belongs in the pydantic issue tracker and not here (FastAPI).

**Comment by Kludex** [comment]
What's your suggestion @srkunze ? :kissing:

**Comment by srkunze** [comment]
As I said, I am new to this, so I would expect you to provide some guidance.

I would find it weird to tell them their purpose.

So, we can leave it as is or ask @samuelcolvin how he would solve this difference in perception.

**Comment by samuelcolvin** [comment]
Short answer: everyone calls it "validation" when they really mean "parsing, validation and corrosion".

Maybe I should stop making the distinction and just suck it up when people complain that pydantic doesn't do what they call validation.

**Comment by srkunze** [comment]
Perhaps using “check plausibility” is what you could substitute it with?

**Comment by srkunze** [comment]
Btw what’s corrosion? Didn’t find anything through a quick search.

**Comment by samuelcolvin** [comment]
Sorry autocorrect, I meant "coercion".

> Perhaps using “check plausibility” is what you could substitute it with?

But pydantic doesn't just "check", it converts.

**Comment by srkunze** [comment]
Coercion sounds to me like converting, modifying, amending.

Checking sounds to me like the part that can be done after a string is converted to a real datatype (the parsing). Only after that it makes sense to check boundaries etc.

So people can continue calling it “validation” (the complete three step process) whereas the middle part is “checking” or “verifying plausibility”.

Makes sense to me at least but do as you see fit. 🙂

**Comment by chbndrhnns** [comment]
IIRC, "coercing" means trying to convert a source type into a target type where both do not match naturally but, for example,  the string "2" can be parsed (coerced) into an integer.
"Validation", on the other hand, would only check that the source type is also an integer and would only proceed if that's the case.

**Comment by kauedg** [comment]
How about changing FastAPI's declaration to something like

>Pydantic is a Python library _used by FastAPI_ to perform data validation.

or yet

>Pydantic is a Python library _used by FastAPI to perform data parsing, validation and coercion_.

That would be more technically precise and would detach the definition from the Pydantic's documentation.
